source/slang/slang-check-decl.cpp

```diff
@@ -1525,29 +1844,72 @@ namespace Slang
         //   but would require synthesizing proxy/forwarding
         //   implementations in the type itself.
         //
-        // We will punt on the second issue for now (since
-        // transparent members aren't currently exposed as
-        // a general-purpose feature for users), and rely
-        // on subsequent checking in this function to
-        // rule out inherited abstract members.
+        // For the first issue, we will use a flag to influence
+        // lookup so that it doesn't include results looked up
+        // through interface inheritance clauses (but it *will*
+        // look up result through inheritance clauses corresponding
+        // to concrete types).
         //
-        auto lookupResult = lookUpMember(m_astBuilder, this, name, type);
+        // The second issue of members that require us to proxy/forward
+        // requests will be handled further down. For now we include
+        // lookup results that might be usable, but not as-is.
+        //
+        auto lookupResult = lookUpMember(m_astBuilder, this, name, type, LookupMask::Default, LookupOptions::IgnoreBaseInterfaces);
+
+        if(!lookupResult.isValid())
+        {
+            // If we failed to even look up a member with the name of the
+            // requirement, then we can be certain that the type doesn't
+            // satisfy the requirement.
+            //
+            // TODO: If we ever allowed certain kinds of requirements to
+            // be inferred (e.g., inferring associated types from the
+            // signatures of methods, as is done for Swift), we'd
+            // need to revisit this step.
+            //
+            getSink()->diagnose(inheritanceDecl, Diagnostics::typeDoesntImplementInterfaceRequirement, type, requiredMemberDeclRef);
+            return false;
+        }
 
         // Iterate over the members and look for one that matches
         // the expected signature for the requirement.
         for (auto member : lookupResult)
         {
+            // To a first approximation, any lookup result that required a "breadcrumb"
```

Comment:
I guess this is another case that `trySynthesizeMethodRequirementWitness` will handle.

Reply:
The hope is that `trySynthesize` can (eventually) handle the arbitrary general case: if your type supports `this.whateverName(whateverArgs...)` syntactically and produces a value of the right type, it will be able to satisfy the `whateverName` requirement.

The loop there is then only responsible for finding "exact match" cases, where the type already has a declaration that is guaranteed to match the exact signature (down to ABI, I guess) of the operation in the interface.

We could in theory eliminate the "exact match" logic if `trySynthesize...` was generalized to handle all kinds of witnesses, at the cost of potentially introducing a bunch of unnecessary synthesized wrapper declarations.